include/deal.II/base/array_view.h

```diff
@@ -39,6 +39,112 @@ class Table;
 template <typename number>
 class LAPACKFullMatrix;
 
+/**
+ * Base class of @p ArrayView which allows strided access into the view.
+ * This is particularly useful when you want to access only one lane of a
+ * VectorizedArray.
+ */
+template <typename ElementType, int stride = 1>
+class StridedArrayView
+{
+public:
+  /**
+   * An alias that denotes the "value_type" of this container-like class,
+   * i.e., the type of the element it "stores" or points to.
+   */
+  using value_type = ElementType;
+
+  /**
+   * Constructor.
+   *
+   * @param[in] starting_element A pointer to the first element of the array
+   * this object should represent.
+   * @param[in] n_elements The length (in elements) of the chunk of memory
+   * this object should represent.
+   *
+   * @note The object that is constructed from these arguments has no
+   * knowledge how large the object into which it points really is. As a
+   * consequence, whenever you call ArrayView::operator[], the array view can
+   * check that the given index is within the range of the view, but it can't
+   * check that the view is indeed a subset of the valid range of elements of
+   * the underlying object that allocated that range. In other words, you need
+   * to ensure that the range of the view specified by the two arguments to
+   * this constructor is in fact a subset of the elements of the array into
+   * which it points. The appropriate way to do this is to use the
+   * make_array_view() functions.
+   */
+  StridedArrayView(value_type *starting_element, const std::size_t n_elements)
+  {
+    this->starting_element = starting_element;
+    this->n_elements       = n_elements;
+  }
```

Comment:
Can you move the definition out of line of the class declaration?

Also, please use the initializer list feature:
```suggestion
  StridedArrayView(value_type *starting_element, const std::size_t n_elements)
    : starting_element (starting_element)
    , n_elements (n_elements)
  {}
```